base/shell/explorer/traywnd.cpp

```diff
@@ -1628,14 +1628,14 @@ class CTrayWindow :
         if (StartSize.cx > rcClient.right)
             StartSize.cx = rcClient.right;
 
-        if (!m_Theme)
+        HWND hwndTaskToolbar = ::GetWindow(m_TaskSwitch, GW_CHILD);
+        if (hwndTaskToolbar)
         {
-            HWND hwndTaskToolbar = ::GetWindow(m_TaskSwitch, GW_CHILD);
-            if (hwndTaskToolbar)
-            {
-                DWORD size = SendMessageW(hwndTaskToolbar, TB_GETBUTTONSIZE, 0, 0);
-                StartSize.cy = HIWORD(size);
-            }
+            LONG EdgeSizeY = GetSystemMetrics(SM_CYEDGE);
+            DWORD size = SendMessageW(hwndTaskToolbar, TB_GETBUTTONSIZE, 0, 0);
+
+            /* Themed button covers Edge area as well */
+            StartSize.cy = HIWORD(size) + (m_Theme ? EdgeSizeY : 0);
```

Comment:
Btw since you use EdgeSizeY only when m_Theme is not NULL, there is no need to systematically call the GetSystemMetrics above in all cases. (that's why in my suggestion I placed that call here directly in place of EdgeSizeY.)

Reply:
Ah, Let me fix this as well. Thanks.

Reply:
OK. Should be done now. Let's see what this looks like. Thanks.